### PR TITLE
Expression evaluator improvements

### PR DESCRIPTION
**Pull request by Kwestor**
This PR contains multiple fixes and new features to recently merged expression evaluator in debugger.

Commit tree represents work done in external repository: feature branches and merge commits with appropriate metadata. Those were all reviewed internally by our team (@pkukielka, @romanowski, @mpociecha and me).

Adds support for:
 - super,
 - named-parameters,
 - nested contructors,
 - nested lambdas,
 - `Array.apply` and rich methods on `Array`s,
 - `isInstanceOf` and `asInstanceOf`,
 - assignment to local fields,
 - varargs all over the place (Java and Scala style, in constructors etc.),
 - accesing `this` in lambdas,
 - refreshing variables view after evaluation,
 - using methods as functions (eta expansion),
 - using methods from objects in lambdas,
 - using values with type params (like `a: List[Int]`) in closures,
 - local implicit values,
 - and more.

Fixes:
 - evaluation of single `val` or `def` does not hang now,
 - printing types now uses `ScalaRunTime.stringOf`, which fixes printing of `Array`s and `String`s (and probably some more),
 - all tests now works on JDK8 (I don't know what fixed it).

For more information, see commit messages or ask.

Only change outside of `expression` modules is to `org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaThread.scala` (last file in diff).

Lastly, the below disclaimer is required by the lawyers:

THE FOLLOWING DISCLAIMER APPLIES TO ALL SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE:
THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.
ONLY THE SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE, IF ANY, THAT ARE ATTACHED TO (OR OTHERWISE ACCOMPANY) THIS SUBMISSION (AND ORDINARY COURSE CONTRIBUTIONS OF FUTURES PATCHES THERETO) ARE TO BE CONSIDERED A CONTRIBUTION. NO OTHER SOFTWARE CODE OR MATERIALS ARE A CONTRIBUTION.

### CODE REVIEW

**Comment by mpociecha** [comment]
You could remove PR numbers since anyway they don't match PR numbers from scala-ide/scala-ide.

**Comment by sschaef** [comment]
add to whitelist

**Comment by Kwestor** [comment]
I removed `#` signs from them.

File: org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaThread.scala [hunk 1]
```diff
@@ -330,8 +334,11 @@ private[model] class ScalaThreadActor private(thread: ScalaThread) extends BaseD
       thread.threadRef.resume()
     case DropToFrame(frame) =>
       thread.dropToFrameInternal(frame)
-    case RebindStackFrames =>
-      if (thread.isSuspended) thread.rebindScalaStackFrames()
+    case RebindStackFrames(shouldFireChangeEvent) =>
+      if (thread.isSuspended) {
+        thread.rebindScalaStackFrames()
+        if (shouldFireChangeEvent) thread.fireChangeEvent(DebugEvent.CONTENT)
```
Comment (dragos) [inline org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaThread.scala]: Is this enough to refresh all the other windows? If whoever triggers this event causes some stack frames to be dropped (I don't know if that's the case), then also the Threads view and the Source editor need to be refreshed.

Comment (mpociecha) [inline org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaThread.scala]: It's only to refresh frames when e.g. someone assigned the new value to var. In EE we don't drop frames.

**Comment by sschaef** [comment]
The commit messages of this PR are not optimal, the `GenChangeLog.bash` script can't handle them properly:
```
[antoras@wookypooky → ~/dev/scala/scala-ide on pr/932]
% ./GenChangeLog.bash 4.0.0-vfinal @

- Revert "Disable 47 tests on Java 8" 
 - # By Piotr Kukiełka
# Via Piotr Kukiełka
* commit '5d8946885768576898bf2c6f2d66a34bc5524cd4':
  Add support for 'super' calls 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '9ce51026ebfba621572aed97072d905077ce294d':
  Throw is and will be unsupported 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '21363cab6525021afb45f206ea3837b8bc4219f1':
  String is not a boxed type
  Changing expected values in tests
  Refactoring primitives 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit 'c4683ca9547a46247f739e68d9406e9ff9db48fb':
  Workaround in TypeCheck removed 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '1455dc3295b390fc044160bf0121ddb1b7672119':
  Support for named parameters 
 - # By Piotr Kukiełka
# Via Piotr Kukiełka
* commit 'f3c0993fc660c8f533b8941958c8317cbf9c1781':
  Allow to access 'this' inside lambdas 
 - # By Michał Pociecha
# Via Michał Pociecha
* commit '4f0d24641dbb251c680d0a33eedc59e6263ed9f3':
  Refresh variables view after evaluation 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '263ce8ab4a93a0b06fefd5beb1095bc446d9077f':
  Implementation moved out of package object 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '417a0f79f51ad00a1db94294c9c2d173f6da642d':
  Change to test evaluation method
  Avoid using `head` on collections
  Fix for printing transformations after error 
 - # By Piotr Kukiełka
# Via Piotr Kukiełka
* commit '6c4758081406d99719575669035c28d62c83c801':
  Add support for using methods from objects as first class functions 
 - # By Piotr Kukiełka
# Via Piotr Kukiełka
* commit 'd2e2b600e05235e37917e8273c2a42302b86683a':
  In case of compilation failure retry with fresh toolbox instance 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit 'c7d272dec047469371ed395b73b66ccd2ad28841':
  Unbound variables and new classes moved to PhaseData
  Time measurement for expression evaluation
  Transforation context introduced 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit 'e4af05ba1986574775474bef915bb1cd73456c90':
  Support for nested constructors 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '8e6fd0a10904d7e7150ff8e010fca30125d29cde':
  Replaces `toString` with `ScalaRunTime.stringOf` 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '1d2ac2eb3c9b31d073b74343e5ab6381055facfb':
  Support for nested Array.apply 
 - * commit 'abc9a6e9feba09c0712915165dd4900ee9f9ab4e':
  Packages now don't have package names. We stop supporting unknown types as lambda parameters. 
 - * commit 'eaf1c6dc826608ed8e5a0001abb010a15426cdef':
  Imports from file are scoped in the same way as in source file. It allows user to use members from imports that are shadowed. 
 - # By Jerzy Müller (1) and Krzysztof Romanowski (1)
# Via Jerzy Müller
* commit 'caa9dfa86d355bc7ff032b05f82235218694f10e':
  Refactoring in TypeNames and NewTypesContext
  Unify all type-to-string convertions and add support for multiple (not all) types of parameter type for lambdas 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '12935dec848d8d0d57a885e2e014c928c6ccff0a':
  Renames `function` to `method` where appropriate 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '926143e40c1e43d121e982ab03f41156c98bbc98':
  More unsupported features detection 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '02a80c3a3fd5c40273dc300c69f53e7753f62613':
  Workaround for single val or def in expression 
 - * commit '3ab52b6be7278b0922bdb90ba504f0a9bec8caea':
  Add support for local implicit values and methods (with nested method limitation) 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '0fd993cce195e5bf6eb744c1cc63551cc4118319':
  Fixes for Stringifier 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '80be10c303a16ecae5cb55203873c8b9cd659023':
  Inform user about not supporting type refinements 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '5855f445ee89584dbe65ef562e7fd6082cc06b57':
  Tests for strange names in all possible places 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '16ab2296afd7632b8fddc314ca93bdd9f0969e25':
  Clean-up of phases code 
 - # By Piotr Kukiełka
# Via Piotr Kukiełka
* commit 'ba54c6f64395e598093a5536553e07979204ca57':
  Add support for Arrays and classOf 
 - # By Piotr Kukiełka
# Via Piotr Kukiełka
* commit '7c7b3ddfa636a079e9cdf9a458d97064e26bcbbf':
  Add support for varargs with two parameter lists 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '007c2a93c48369515607bcf9c3e04f1e71b65583':
  Support for assignment to local fields 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '1f17c21353f4fced4dd683ba5a4f84c33b67e593':
  Remove superfluous () after variables section 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '5e270d94d86256ebedfb5cec239ac4b48f9ea9b5':
  Support for `asInstanceOf`
  Support for `isInstanceOf` 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit 'dfb2fb094ecb45704a05da41fbb87a1b56b48d40':
  Another dead code cleanup after stub removal 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit '9d63707fb1426aaaedba8eb1549f751becf69f9e':
  Remove JdiProxyWrapper dead code 
- Implement partial support for nested methods. We don't support closures and some nesting configuration where we cannot determine which method should be pick. We inform user about this. 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit 'ca0e1f0d155cb2022ec2bfeff3f4425910c0b5de':
  Tests for for comprehensions (and loops) 
 - # By Piotr Kukiełka
# Via Piotr Kukiełka
* commit 'ffaf55ffe31f1abd7a2ef5823a9381ffdcfd7c2f':
  Restore code removed during merge
  Split MethodsAsFunctionsTest into separate files 
- Remove abstract methods from candidates to invoke 
- Whitespace 
 - # By Jerzy Müller
# Via Jerzy Müller
* commit 'c0bd0085cb80cb907fce19a18410a5ec166da8ee':
  Review fixes
  Error handling moved from ExprMan to ExprExc
  Move compiled code to JdiExpression
  Support for Java-style varargs
  Support for Java setters/getters
  Support for varargs in Scala constructors
  Cosmetic changes to JdiMethodInvoker
  Refactoring JdiMethodInvoker
  Better error reporting in tests 
 - * commit 'f979269e57af99ad92a63e5fd3cc0fbda8d28edf':
  Nested lambdas. It just load that nested ones to remote jvm 
 - * commit 'c27d77cb9fb93cbde1609d53230c56320eb9d7fd':
  add tests - it appears to work 
 - # By Piotr Kukiełka
# Via Piotr Kukiełka
* commit '8e3a2aa4be0a2d8ccc5942ec7df9ecc51ad787ad':
  Add proper support for implicit classes which extends AnyVal 
 - * commit 'e79379f3692586701c166d06e7b629a2b7ccbe7d':
  Add support for using methods from scope as first class functions 
 - Revert "Mark several tests as flaky" 
 - Mark several tests as flaky 
 - Basic support for Hot Code Replacement in debugged JVM 
 - Add Expression Evaluator to update site 
 - Small Scaladoc corrections 
 - mark occurrences improvements (:ticket:`1002417` :ticket:`1002418`)
 - Expression evaluator 
 - T1002180 remove old wizards (:ticket:`1002180`)
 - Also apply refactoring to files not currently open in the editor (:ticket:`1002079`)
 - Make organize imports format consistent with scalariform 
 - old version number and transitive deps 
 - Handle single empty line in `RemoveDuplicatedEmptyLines` save action (:ticket:`1002400`)
 - Take care of file rename operations in `rename` refactorings (:ticket:`1001928`)
 - Fix NPE in Scala Reconciler (:ticket:`1002396`)
 - Fix warnings 
 - Mark some tests as flaky 
 - Improve wording in implicit preference page (:ticket:`1001003`)
 - Don't leak `ScalaSourceFileEditors` 
 - Update URL of equinox weaving launcher 
 - T1002387 Remove an option to close SPCs when editors are open 
 - Improvements for NamePrinter and NamePrinterTest 
 - Prevent race condition in CallByNameParamAtCreationPresenter (:ticket:`1002388`)
 - Add command extension to restart presentation compiler with shortcut (:ticket:`1002391`)
 - Enable retry for a seemingly flaky test. 
 - Fix NPE in semantic highlighting job (:ticket:`1002386`)
 - No need to run `FreshRunReq`, this is normal operation for the PC. 
 - Handle auto indent correctly after unicode representation of => (:ticket:`1002383`)
 - Handle partially applied functions properly (:ticket:`1002381`)
 - Highlight by-name parameters at their creation (:ticket:`1002340`)
 - Correct debug messages in ScalaPlugin 
 - Add a Gitter chat badge to README.md 
 - Bump version to new Scala 2.11.6 development version, including xml and 
 - Check return value of getResource call to avoid NPE (:ticket:`1002375`)
 - Update editor only once after execution of save actions (:ticket:`1002309`)
 - Add strip margins in string interpolation  (:ticket:`1002145`)
 - Throwing OCE if build was interrupted (:ticket:`1002365`)
 - Better messages for custom annotations 
 - Fix double-newline bug in Scala templates (:ticket:`1002303`)
 - Fixes StackOverflowError in `RegionUtils.subtract` (:ticket:`1002361`)
 - Fixes several warnings that occur during the build 
 - show svelto link in report bug dialog 
 - Fix spelling mistake in constant in SdtConstants 
 - Implementation for "Copy Qualified Name" (#1001257) (:ticket:`1001257` :ticket:`1001257`)
 - Remove version numbers that made the plugin refuse to install in Kepler. 
 - Correctly handle NullaryMethodTypes in the java type mapper (:ticket:`1001734`)
 - Create the right ElementInfo for package declaration. (:ticket:`1000572`)
 - Partial revert of d9fbd270 (:ticket:`1002355`)
 - Fixes race condition during hyperlinking (:ticket:`1002352`)
 - Bump development version to 4.1.0 
 - Support for drop to frame 
 - New File Wizard works with projects whose source dir is the root dir (:ticket:`1002332`)
 - Faster project loading in classpath computation (:ticket:`1002179` :ticket:`1002311`)
 - Remove instance-creating implicit that shadowed BooleanSettingWithDefault 
 - code completion: reduce relevance for non class fields and for java.*, scala.* packages (:ticket:`1002343`)
 - Revert "Treat files as external dependencies." 
 - Treat files as external dependencies. (:ticket:`1002344`)
 - Automatically select problem position on quick assist invocation (:ticket:`1002305`)
 - Adds back Scala notation in the outline labels (:ticket:`1002339`)
 - Quick fix exceptions 
 - scaliform default preferences changed (:ticket:`1002321`)
 - Handle src/bin folders that are the same as project root folders (:ticket:`1002146`)
 - Highlight by name parameters at call site (#1000989) (:ticket:`1000989`)
 - Remove `o.s.u.ui.SWTUtils` (:ticket:`1002288`)
 - Add quick assists for spelling mistakes (:ticket:`1002301`)
 - Consider project specific formatting settings for auto format (:ticket:`1002331`)
 - Mark a recently flaky test as, erm, Flaky. 
 - Make the build work on JDK 8 
 - Updates to Scala 2.11.5-SNAPSHOT as default version
```

My auto-edits branch can't be handled either. I guess the easiest way to fix it is to do it manually.

**Comment by sschaef** [comment]
LGTM

**Comment by sschaef** [comment]
The last commit fixes [#1002428](https://www.assembla.com/spaces/scala-ide/tickets/1002428#/activity/ticket:)

**Comment by Kwestor** [comment]
I've added a simple change to way values are checked in tests.

@sschaef do you want me to add `Fix #1002428` to commit description?

**Comment by sschaef** [comment]
> do you want me to add Fix #1002428 to commit description?

Yeah, would be nice.

**Comment by sschaef** [comment]
I found the following race condition before this PR was merged:
```
2015-04-28 23:33:14,661 ERROR [Worker-109] - ExpressionManager$ - Unknown exception during evaluation
java.lang.AssertionError: assertion failed: Race condition detected: You are running a presentation compiler method outside the PC thread.[phase: <no phase>] Please file a ticket with the current stack trace at https://www.assembla.com/spaces/scala-ide/support/tickets
	at scala.tools.nsc.Global.assert(Global.scala:262)
	at scala.tools.nsc.interactive.Global.assertCorrectThread(Global.scala:535)
	at scala.reflect.internal.Symbols$Symbol.rawInfo(Symbols.scala:1565)
	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1499)
	at scala.reflect.internal.Symbols$Symbol.tpe_$times(Symbols.scala:1453)
	at scala.reflect.internal.Symbols$Symbol.isErroneous(Symbols.scala:781)
	at scala.reflect.internal.Printers$class.symNameInternal(Printers.scala:38)
	at scala.reflect.internal.Printers$class.symName(Printers.scala:47)
	at scala.reflect.internal.SymbolTable.symName(SymbolTable.scala:16)
	at scala.reflect.internal.Printers$class.backquotedPath(Printers.scala:54)
	at scala.reflect.internal.SymbolTable.backquotedPath(SymbolTable.scala:16)
	at scala.reflect.internal.Printers$TreePrinter$$anonfun$printTree$9.apply(Printers.scala:357)
	at scala.reflect.internal.Printers$TreePrinter$$anonfun$printTree$9.apply(Printers.scala:357)
	at scala.reflect.internal.Printers$TreePrinter.printImport(Printers.scala:261)
	at scala.reflect.internal.Printers$TreePrinter.printTree(Printers.scala:357)
	at scala.tools.nsc.ast.Printers$TreePrinter$$anonfun$print$1.apply(Printers.scala:20)
	at scala.tools.nsc.ast.Printers$TreePrinter$$anonfun$print$1.apply(Printers.scala:17)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)
	at scala.tools.nsc.ast.Printers$TreePrinter.print(Printers.scala:17)
	at scala.reflect.api.Printers$class.render(Printers.scala:185)
	at scala.reflect.api.Universe.render(Universe.scala:61)
	at scala.reflect.api.Printers$class.show(Printers.scala:201)
	at scala.reflect.api.Universe.show(Universe.scala:61)
	at scala.reflect.api.Printers$class.treeToString(Printers.scala:193)
	at scala.reflect.api.Universe.treeToString(Universe.scala:61)
	at scala.reflect.api.Trees$TreeApi$class.toString(Trees.scala:178)
	at scala.reflect.internal.Trees$TreeContextApiImpl.toString(Trees.scala:102)
	at org.scalaide.debug.internal.expression.sources.Imports$ImportExtractor$ImportTraverser$$anonfun$apply$2.apply(Imports.scala:63)
	at org.scalaide.debug.internal.expression.sources.Imports$ImportExtractor$ImportTraverser$$anonfun$apply$2.apply(Imports.scala:63)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:245)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:245)
	at scala.collection.immutable.List.map(List.scala:285)
	at org.scalaide.debug.internal.expression.sources.Imports$ImportExtractor$ImportTraverser.apply(Imports.scala:63)
	at org.scalaide.debug.internal.expression.sources.Imports$ImportExtractor.importForFile(Imports.scala:68)
	at org.scalaide.debug.internal.expression.sources.Imports$.org$scalaide$debug$internal$expression$sources$Imports$$getImports(Imports.scala:77)
	at org.scalaide.debug.internal.expression.sources.Imports$$anonfun$importsFromCurrentStackFrame$1.apply(Imports.scala:81)
	at org.scalaide.debug.internal.expression.sources.Imports$$anonfun$importsFromCurrentStackFrame$1.apply(Imports.scala:81)
	at org.scalaide.debug.internal.expression.sources.SPCIntegration$$anonfun$forCurrentStackFrame$2$$anonfun$apply$1.apply(SPCIntegration.scala:32)
	at org.scalaide.debug.internal.expression.sources.SPCIntegration$$anonfun$forCurrentStackFrame$2$$anonfun$apply$1.apply(SPCIntegration.scala:31)
	at org.scalaide.core.internal.compiler.PresentationCompilerProxy$$anonfun$internal$1.apply(PresentationCompilerProxy.scala:105)
	at org.scalaide.core.internal.compiler.PresentationCompilerProxy$$anonfun$internal$1.apply(PresentationCompilerProxy.scala:104)
	at scala.Option.flatMap(Option.scala:171)
	at org.scalaide.core.internal.compiler.PresentationCompilerProxy.internal(PresentationCompilerProxy.scala:104)
	at org.scalaide.core.internal.compiler.PresentationCompilerProxy.apply(PresentationCompilerProxy.scala:62)
	at org.scalaide.debug.internal.expression.sources.SPCIntegration$$anonfun$forCurrentStackFrame$2.apply(SPCIntegration.scala:31)
	at org.scalaide.debug.internal.expression.sources.SPCIntegration$$anonfun$forCurrentStackFrame$2.apply(SPCIntegration.scala:30)
	at scala.Option.fold(Option.scala:158)
	at org.scalaide.debug.internal.expression.sources.SPCIntegration$class.forCurrentStackFrame(SPCIntegration.scala:30)
	at org.scalaide.debug.internal.expression.sources.Imports$.forCurrentStackFrame(Imports.scala:11)
	at org.scalaide.debug.internal.expression.sources.Imports$.importsFromCurrentStackFrame(Imports.scala:81)
	at org.scalaide.debug.internal.expression.proxies.phases.AddImports.transform(PackInFunction.scala:37)
	at org.scalaide.debug.internal.expression.ExpressionEvaluator$$anonfun$1.apply(ExpressionEvaluator.scala:150)
	at org.scalaide.debug.internal.expression.ExpressionEvaluator$$anonfun$1.apply(ExpressionEvaluator.scala:144)
	at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
	at scala.collection.immutable.List.foldLeft(List.scala:84)
	at org.scalaide.debug.internal.expression.ExpressionEvaluator.org$scalaide$debug$internal$expression$ExpressionEvaluator$$transform(ExpressionEvaluator.scala:144)
	at org.scalaide.debug.internal.expression.ExpressionEvaluator$$anonfun$compileExpression$1.apply(ExpressionEvaluator.scala:78)
	at org.scalaide.debug.internal.expression.ExpressionEvaluator$$anonfun$compileExpression$1.apply(ExpressionEvaluator.scala:71)
	at scala.util.Try$.apply(Try.scala:191)
	at org.scalaide.debug.internal.expression.ExpressionEvaluator.compileExpression(ExpressionEvaluator.scala:71)
	at org.scalaide.debug.internal.expression.JdiExpressionEvaluator.apply(JdiExpressionEvaluator.scala:60)
	at org.scalaide.debug.internal.expression.ExpressionManager$class.computeInEvaluator$1(ExpressionManager.scala:99)
	at org.scalaide.debug.internal.expression.ExpressionManager$$anonfun$compute$1.apply(ExpressionManager.scala:127)
	at org.scalaide.debug.internal.expression.ExpressionManager$$anonfun$compute$1.apply(ExpressionManager.scala:124)
	at org.scalaide.debug.internal.expression.ExpressionManager$EvaluationStatus$.monitor(ExpressionManager.scala:77)
	at org.scalaide.debug.internal.expression.ExpressionManager$class.compute(ExpressionManager.scala:124)
	at org.scalaide.debug.internal.expression.ExpressionManager$.compute(ExpressionManager.scala:50)
	at org.scalaide.debug.internal.ui.ExpressionEvaluatorView$$anon$1.run(ExpressionEvaluatorView.scala:83)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:54)
```
Can someone please verify if it can still occur on master?

**Comment by Kwestor** [comment]
I remember fixing some race condition involving PC, but I don't remember if it was this one.

**Pull request merged**